engine/src/main/java/org/terasology/rendering/world/selection/BlockSelectionRenderer.java

```diff
@@ -130,7 +129,7 @@ public void renderMark2(Vector3ic blockPos) {
     }
 
     private Vector3f getCameraPosition() {
-        return JomlUtil.from(CoreRegistry.get(WorldRenderer.class).getActiveCamera().getPosition());
+        return CoreRegistry.get(WorldRenderer.class).getActiveCamera().getPosition();
```

Comment:
This directly returns the camera position in a way that it could (accidentally) be modified. I'd suggest to change the method's return type to `Vector3fc`.

Reply:
It seems that in `ShadowMapNode.java` in `CoreRendering`, this is intentionally used.
See https://github.com/Terasology/CoreRendering/blob/develop/src/main/java/org/terasology/corerendering/rendering/dag/nodes/ShadowMapNode.java#L210 and https://github.com/Terasology/CoreRendering/blob/develop/src/main/java/org/terasology/corerendering/rendering/dag/nodes/ShadowMapNode.java#L215